### PR TITLE
Do not consider .ignore files when searching with ripgrep

### PR DESCRIPTION
**Pull request by 183amir**
Under some special conditions, for example installing miniconda in a folder that contains `.gitignore`, behavior of ripgrep can change during builds.
Adding the `--unrestricted` flag avoids this.
```
    -u, --unrestricted                      
            Reduce the level of "smart" searching. A single -u won't respect .gitignore
            (etc.) files. Two -u flags will additionally search hidden files and
            directories. Three -u flags will additionally search binary files.
            
            'rg -uuu' is roughly equivalent to 'grep -r'.
```

### CODE REVIEW

**Comment by tiagofrepereira2012** [comment]
Hi, follow a small example to see the unexpected behaviour aforementioned.

```
git clone https://github.com/tiagofrepereira2012/dummy && \
cd dummy && \
wget https://repo.anaconda.com/miniconda/Miniconda3-4.7.10-Linux-x86_64.sh && \
bash ./Miniconda3-4.7.10-Linux-x86_64.sh -b -p ./miniconda && \
source ./miniconda/etc/profile.d/conda.sh && \
conda activate base && \
conda install -n base python conda-build=3 conda=4 conda-verify=3 --yes && \
conda build ./conda --no-anaconda-upload -m build_config.yaml
```

My dummy project contains a `.gitignore` file containing the directory `bin`.
Then, I download the 'miniconda` and install the latest `conda-build`.
My dummy recipe has a entry-point called `dummy-print.py` in the test stage.
Finally, when i do the `conda build` I have the following error during the test of the recipe:

```
## Package Plan ##

  environment location: /build_scripts/dummy/miniconda/conda-bld/dummy_1570636067408/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl


The following NEW packages will be INSTALLED:

    _libgcc_mutex:   0.1-main                defaults
    ca-certificates: 2019.8.28-0             defaults
    certifi:         2019.9.11-py36_0        defaults
    dummy:           0.0.1-py36he61d863_0    local   
    libedit:         3.1.20181209-hc058e9b_0 defaults
    libffi:          3.2.1-hd88cf55_4        defaults
    libgcc-ng:       9.1.0-hdf63c60_0        defaults
    libstdcxx-ng:    9.1.0-hdf63c60_0        defaults
    ncurses:         6.1-he6710b0_1          defaults
    openssl:         1.1.1d-h7b6447c_2       defaults
    python:          3.6.9-h265db76_0        defaults
    readline:        7.0-h7b6447c_5          defaults
    setuptools:      41.4.0-py36_0           defaults
    sqlite:          3.30.0-h7b6447c_0       defaults
    tk:              8.6.8-hbc83047_0        defaults
    xz:              5.2.4-h14c3975_4        defaults
    zlib:            1.2.11-h7b6447c_3       defaults

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
export PREFIX=/build_scripts/dummy/miniconda/conda-bld/dummy_1570636067408/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl
export SRC_DIR=/build_scripts/dummy/miniconda/conda-bld/dummy_1570636067408/test_tmp
import: 'dummy'
import: 'dummy'
+ dummy_print.py
/build_scripts/dummy/miniconda/conda-bld/dummy_1570636067408/test_tmp/run_test.sh: /build_scripts/dummy/miniconda/conda-bld/dummy_1570636067408/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/dummy_print.py: /build_scripts/dummy/miniconda/conda-bld/dummy_1570636067408/_h_env_placehold_: bad interpreter: No such file or directory
Tests failed for dummy-0.0.1-py36he61d863_0.tar.bz2 - moving package to /build_scripts/dummy/miniconda/conda-bld/broken
WARNING:conda_build.build:Tests failed for dummy-0.0.1-py36he61d863_0.tar.bz2 - moving package to /build_scripts/dummy/miniconda/conda-bld/broken
WARNING conda_build.build:tests_failed(2230): Tests failed for dummy-0.0.1-py36he61d863_0.tar.bz2 - moving package to /build_scripts/dummy/miniconda/conda-bld/broken
TESTS FAILED: dummy-0.0.1-py36he61d863_0.tar.bz2
```

Basically, if you `--trace` what ripgrep is doing here `https://github.com/conda/conda-build/blob/3.18.9/conda_build/build.py#L203` you will see that it takes into account `.gitignore` file and ignores everything inside of `./bin`.

**Comment by marcelotrevisani** [comment]
Hi @183amir ,
Thanks for your contribution! Could you please just add a test to cover this scenario? So in that way, if someone removes this flag in the future we will be able to track it easily.

**Comment by 183amir** [comment]
Hi @marcelotrevisani, Thank you for considering this pull request. I have considered adding a test case here, but I am afraid the test setup to trigger this bug is quite complicated. It took us weeks to figure out what is going on and 2 days to create a reproducible easy example to showcase this as done by @tiagofrepereira2012. To trigger this bug, miniconda needs to be installed in a special folder with a `.gitignore` file. I don't think that is possible to test in the current setup of cona-build tests, is it?

**Comment by marcelotrevisani** [comment]
> Hi @marcelotrevisani, Thank you for considering this pull request. I have considered adding a test case here, but I am afraid the test setup to trigger this bug is quite complicated. It took us weeks to figure out what is going on and 2 days to create a reproducible easy example to showcase this as done by @tiagofrepereira2012. To trigger this bug, miniconda needs to be installed in a special folder with a `.gitignore` file. I don't think that is possible to test in the current setup of cona-build tests, is it?

Did you install miniconda in a non-empty folder?
I thought the problem was because your project had a ``.gitignore`` during the building phase

**Comment by tiagofrepereira2012** [comment]
Hi @marcelotrevisani,

We found this issue while running `conda build..` in our ci.
In this case, the miniconda is installed in the project directory, which is a git cloned repo. Therefore we have a `.gitgnore` file.

**Comment by tiagofrepereira2012** [comment]
In my opinion, the default behaviour of `have_prefix_files` function should allow you to search in all files with no restrictions (in favor of @183amir PR).
The choice to ignore some files, if needed, should be stored explicitly in an argument of this function.

It was really hard to figured was was going on with this obscured behaviour of `ripgrep`.

**Comment by marcelotrevisani** [comment]
> Hi @marcelotrevisani,
> 
> We found this issue while running `conda build..` in our ci.
> In this case, the miniconda is installed in the project directory, which is a git cloned repo. Therefore we have a `.gitgnore` file.

Could you please describe how your project is structured and where conda is installed and things like that?
There is a way to cover it with tests, I may take a look into it in the next days.

**Comment by 183amir** [comment]
Hi Marcelo, the steps to reproduce the issue are provided in Tiago's first comment. Is that enough?

**Comment by tiagofrepereira2012** [comment]
Hi @marcelotrevisani,

You can run the steps bellow to reproduce the issue.
Basically, this is a summarization of the steps that our CI runs when build a conda package.
Note that the miniconda is installed in the same folder as the project.
This shouldn't be a problem. The problem is `conda build` ignoring some files without letting you know (thanks to the default behaviour of `ripgrep`) and the only way to know is by debugging.

```
git clone https://github.com/tiagofrepereira2012/dummy && \
cd dummy && \
wget https://repo.anaconda.com/miniconda/Miniconda3-4.7.10-Linux-x86_64.sh && \
bash ./Miniconda3-4.7.10-Linux-x86_64.sh -b -p ./miniconda && \
source ./miniconda/etc/profile.d/conda.sh && \
conda activate base && \
conda install -n base python conda-build=3 conda=4 conda-verify=3 --yes && \
conda build ./conda --no-anaconda-upload -m build_config.yaml
```

**Comment by tiagofrepereira2012** [comment]
Thanks @marcelotrevisani
